src/OrbitBase/include/OrbitBase/FutureHelpers.h

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_BASE_FUTURE_HELPERS_H_
+#define ORBIT_BASE_FUTURE_HELPERS_H_
+
+#include <type_traits>
+#include <utility>
+
+#include "OrbitBase/Future.h"
+#include "OrbitBase/Promise.h"
+
+namespace orbit_base {
+
+template <typename T, typename Invocable>
+void RegisterContinuationOrCallDirectly(const Future<T>& future, Invocable continuation) {
+  const auto result = future.RegisterContinuation(continuation);
+
+  if (result == FutureRegisterContinuationResult::kFutureAlreadyCompleted) {
+    continuation(future.Get());
+  }
+}
+
+template <typename Invocable>
+void RegisterContinuationOrCallDirectly(const Future<void>& future, Invocable continuation) {
+  const auto result = future.RegisterContinuation(continuation);
+
+  if (result == FutureRegisterContinuationResult::kFutureAlreadyCompleted) {
+    continuation();
+  }
+}
+
+template <typename T>
+[[nodiscard]] inline Future<T> UnwrapFuture(const Future<T>& future) {
+  return future;
+}
+
+template <typename T>
+[[nodiscard]] inline Future<T> UnwrapFuture(const Future<Future<T>>& outer_future) {
+  CHECK(outer_future.IsValid());
+
+  auto promise = std::make_shared<Promise<T>>();
+
+  auto outer_continuation = [promise](const Future<T>& inner_future) {
```

Comment:
maybe short explanation of what is happening here. I spent some time unwrapping this code :)

Reply:
Yes, that's a very good idea. I also needed some time to figure out the correct implementation. Unit tests to the rescue. :-)

Reply:
done.